### ISSUE TITLE
Cannot find module '@openauthjs/openauth/client' or its corresponding type declarations

### ISSUE THREAD

**Issue by fredericrous** [openauthjs/openauth#113]
reproduction workflow

create an expo app

```console
npx create-expo-app@latest
```

for info, the tsconfig of the expo app will be https://github.com/expo/expo/blob/main/packages/expo/tsconfig.base.json

now import createClient from 

```ts
import { createClient } from "@openauthjs/openauth/client"
```

like documented at https://github.com/openauthjs/openauth?tab=readme-ov-file#auth-client

You will get the error

```
Cannot find module '@openauthjs/openauth/client' or its corresponding type declarations
```

workaround

```ts
// @ts-expect-error -- type issue
import { createClient } from '@openauthjs/openauth/dist/esm/client'
```

**Comment by fredericrous** [comment]
apparently I'm not the only one. a user found a workaround, which is to copy the client in a local context, along with the files it depends on:
- session
- utils
- error
- pkce

 https://gist.github.com/xinha-sh/723fae6bad78a303bbaeeb2b41634c76

**Comment by vitostamatti** [comment]
any improvements on this?? i dont like the copy/paste approach

**Comment by fredericrous** [comment]
forking the client is an approach hard to maintain
I tried to ditch OpenAUTH client and use instead Expo AuthSession, however I gave up when I noticed that when calling [AuthSession.fetchDiscoveryAsync(issuer)](https://docs.expo.dev/versions/latest/sdk/auth-session/#authsessionfetchdiscoveryasyncissuer), the url called was hardcoded to [.well-known/openid-configuration](https://github.com/expo/expo/blob/33ee7eb45db3554835cd4a71598300f8de9f71fe/packages/expo-auth-session/src/Discovery.ts#L211) and openauth uses [.well-known/oauth-authorization-server](https://github.com/openauthjs/openauth/blob/b6ecb636e8d0aaaede63dc6a5f2382bd5bd2cc87/packages/openauth/src/issuer.ts#L396)

**Comment by vitostamatti** [comment]
i managed to make it work using AuthSession by manually providing the urls for discovery .

```javascript
const discovery = {
    authorizationEndpoint: "your.auth.domain/authorize",
    tokenEndpoint: "your.auth.domain/token"
}
const [ request, response, promptAsync] = useAuthRequest(
    {
        clientId:"your-mobile-app",
        redirectUri: makeRedirectUri() // from expo-auth-session
    },
    discovery
) 
```

Have you try [this option](https://metrobundler.dev/docs/package-exports) on metrobundler ?

**Comment by xinha-sh** [comment]
> i managed to make it work using AuthSession by manually providing the urls for discovery .
> 
> 
> 
> ```javascript
> 
> const discovery = {
> 
>     authorizationEndpoint: "your.auth.domain/authorize",
> 
>     tokenEndpoint: "your.auth.domain/token"
> 
> }
> 
> const [ request, response, promptAsync] = useAuthRequest(
> 
>     {
> 
>         clientId:"your-mobile-app",
> 
>         redirectUri: makeRedirectUri() // from expo-auth-session
> 
>     },
> 
>     discovery
> 
> ) 
> 
> ```
> 
> 
> 
> Have you try [this option](https://metrobundler.dev/docs/package-exports) on metrobundler ?
> 
> 

Can you share the complete code ? Like how would a code adapter flow work with this ?

**Comment by mib200** [comment]
Why not make `.well-known/openid-configuration` to be used as the config endpoint? Reason being, lot of services like AWS HTTP authorizer etc only take base issuer URL, and append the path to search of openid configuration, the non standard URL just makes it break.

**Comment by fredericrous** [comment]
> i managed to make it work using AuthSession by manually providing the urls for discovery .
> 
> const discovery = {
>     authorizationEndpoint: "your.auth.domain/authorize",
>     tokenEndpoint: "your.auth.domain/token"
> }
> const [ request, response, promptAsync] = useAuthRequest(
>     {
>         clientId:"your-mobile-app",
>         redirectUri: makeRedirectUri() // from expo-auth-session
>     },
>     discovery
> ) 
> Have you try [this option](https://metrobundler.dev/docs/package-exports) on metrobundler ?

yes! I got it working this way, thanks. I let the issue open because imo we should at least document that for Expo we should use useAuthRequest instead of @openauthjs/openauth/client
for people looking for an example, look at https://docs.expo.dev/guides/authentication/#calendly

**Comment by moAlobaidi** [comment]
@fredericrous  you can import like this to fix:

```import { createClient } from "@openauthjs/openauth/src/client";```

**Comment by fredericrous** [comment]
@vitostamatti did you get `useAuthRequest` working on Android? I'm hitting https://github.com/expo/expo/issues/12044

**Comment by Ruud-BMair** [comment]
@fredericrous Could you tell me how you got it working? 
I'm still getting `{"type": "dismiss"}` as response on Android.
I'm using `com.[company].[app].dev` as `scheme` and as `android.package`
And in the authrequest I'm using `responseType: "token"`

**Comment by fredericrous** [comment]
I literally started from the example provided in PR #222 I made sure the example worked. Then I edited the scheme and package name. tested the example was still working. Then I copied over my files onto the example, iteratively. I often tested things didn't break. I don't remember more than that

**Comment by Ruud-BMair** [comment]
@fredericrous Thank you for your help. I fixed it, I'm not entirely sure what the issue was. I kept my scheme the same and I made a seperate screen for login: sign-in and made the redirect uri path `sign-in`
